File: app_java/src/main/java/tech/thdev/android_mvp_sample/view/main/presenter/MainContract.java
package tech.thdev.android_mvp_sample.view.main.presenter;

import android.content.Context;

import tech.thdev.android_mvp_sample.adapter.contract.ImageAdapterContract;
import tech.thdev.android_mvp_sample.data.SampleImageData;

/**
 * Created by tae-hwan on 12/22/16.
 */
public interface MainContract {

    interface View {

    }

    interface Presenter {

        void attachView(View view);

        void setImageAdapterModel(ImageAdapterContract.Model adapterModel);

        void setImageAdapterView(ImageAdapterContract.View adapterView);

        void detachView();

        void setSampleImageData(SampleImageData sampleImageData);

        void loadItems(Context context, boolean isClear);
    }
}

File: app_kotlin/src/main/java/tech/thdev/app_kotlin/view/main/presenter/MainContract.kt
package tech.thdev.app_kotlin.view.main.presenter

import android.content.Context

/**
 * Created by tae-hwan on 12/23/16.
 */
interface MainContract {

    interface View {

        fun showToast(title: String)
    }

    interface Presenter {
        fun loadItems(context: Context, isClear: Boolean)
    }
}